packages/mui-system/src/color/core.ts

```diff
@@ -0,0 +1,84 @@
+import { get, set } from './bit';
+
+/* eslint-disable no-bitwise */
+/* eslint-disable prefer-template */
+
+export type Color = number;
+
+export const OFFSET_R = 24;
+export const OFFSET_G = 16;
+export const OFFSET_B = 8;
+export const OFFSET_A = 0;
+
+export function newColor(r: number, g: number, b: number, a: number) {
+  // prettier-ignore
+  return (
+    (r << OFFSET_R) +
+    (g << OFFSET_G) +
+    (b << OFFSET_B) +
+    (a << OFFSET_A)
+  );
+}
+
+export function from(hex: number) {
+  return newColor(get(hex, OFFSET_R), get(hex, OFFSET_G), get(hex, OFFSET_B), get(hex, OFFSET_A));
+}
+
+export function getRed(c: Color) {
+  return get(c, OFFSET_R);
+}
+export function getGreen(c: Color) {
+  return get(c, OFFSET_G);
+}
+export function getBlue(c: Color) {
+  return get(c, OFFSET_B);
+}
+export function getAlpha(c: Color) {
+  return get(c, OFFSET_A);
+}
+
+export function setRed(c: Color, value: number) {
+  return set(c, OFFSET_R, value);
+}
+export function setGreen(c: Color, value: number) {
+  return set(c, OFFSET_G, value);
+}
+export function setBlue(c: Color, value: number) {
+  return set(c, OFFSET_B, value);
+}
+export function setAlpha(c: Color, value: number) {
+  return set(c, OFFSET_A, value);
+}
+
+/**
+ * Map 8-bits value to its hexadecimal representation
+ * ['00', '01', '02', ..., 'fe', 'ff']
+ */
+const FORMAT_HEX = Array.from({ length: 256 }).map((_, byte) => byte.toString(16).padStart(2, '0'));
+
+export function format(color: Color): string {
```

Comment:
```suggestion
export function formatHex(color: Color): string {
```
Is there a reason not to use `formatHex`?

Reply:
I meant for hex to be the default representation, so I want to guide users of this module towards that function.